webapp/lib/controller.dart

```diff
@@ -4,7 +4,102 @@ import 'logger.dart';
+}
+
+class ConfigurationResponseData extends Data {
+  String parentTag;
+  String language;
+  String text;
+  int index;
+  ConfigurationResponseData({this.parentTag, this.index, this.language, this.text});
+}
+
+Map<String, Map<String, List<String>>> configurationTagData;
+Set<String> additionalConfigurationTags;
+Set<String> configurationResponseLanguages;
+
+void init() async {
+  view.init();
+  await platform.init();
+}
+
+void initUI() {
+  view.contentView.dashboardView.activePackages.addAll(
+    [
+      new view.ActivePackagesViewPartial('Urgent conversations'),
+      new view.ActivePackagesViewPartial('Open conversations'),
+      new view.ActivePackagesViewPartial('Batch replies (Week 12)'),
+    ]);
+  view.contentView.dashboardView.availablepackages.addAll(
+    [
+      new view.AvailablePackagesViewPartial('Quick Poll',
+        'Ask a question with fixed answers',
+        ['Needs: Q/A, Labelling team, Safeguarding response', 'Produces: Dashboard for distribution of answers']),
+      new view.AvailablePackagesViewPartial('Information Service',
+        'Answer people\'s questions',
+        ['Needs: Response protocol, Labelling team, Safeguarding response', 'Produces: Thematic distribution, work rate tracker']),
+      new view.AvailablePackagesViewPartial('Bulk Message',
+        'Send set of people a once off message',
+        ['Needs: Definition of who. Safeguarding response', 'Produces: Success/Fail tracker'])
+    ]);
+  view.contentView.dashboardView.renderActivePackages();
+  view.contentView.dashboardView.renderAvailablePackages();
+  view.contentView.renderView(view.contentView.dashboardView.dashboardViewElement);
+}
+
+void command(UIAction action, Data actionData) {
+  log.verbose('command => $action : $actionData');
+  switch (action) {
+
+    case UIAction.userSignedIn:
+      initUI();
+      break;
+    case UIAction.goToConfigurator:
+      fetchConfigurationData();
+      showConfigurationView(null, configurationTagData);
+      break;
+    case UIAction.configurationTagSelected:
+      ConfigurationTagData data = actionData;
+      showConfigurationView(data.selectedTag, configurationTagData);
+      break;
+    case UIAction.addConfigurationTag:
+      ConfigurationTagData data = actionData;
+      addNewConfigurationTag(data.tagToAdd, configurationResponseLanguages, additionalConfigurationTags, configurationTagData);
+      break;
+    case UIAction.editConfigurationTagResponse:
+      ConfigurationResponseData data = actionData;
+      updateEditedConfigurationTagResponse(data.parentTag, data.index, data.language, data.text);
+      break;
+    case UIAction.addConfigurationResponseEntries:
+      ConfigurationResponseData data = actionData;
+      addConfigurationResponseEntries(data.parentTag, data.language, data.text, configurationTagData);
+      break;
+  }
+}
+
+Set<String> get configurationReponseLanguageData { return {'English', 'Somali'};}
```

Comment:
you can also rewrite this to use the `=>` which is the dart recommended way of defining single line functions:
```suggestion
List<String> get configurationReponseLanguageData => ['English', 'Somali'];
```